src/System.ComponentModel.Composition/src/Microsoft/Internal/BinaryCompatability.cs

```diff
@@ -0,0 +1,63 @@
+//------------------------------------------------------------------------------
+// <copyright file="BinaryCompatibility.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>                                                                
+//------------------------------------------------------------------------------
+using System;
+using System.Diagnostics.Contracts;
+using System.Reflection;
+using System.Runtime.Versioning;
+using System.Security;
+
+namespace Microsoft.Internal
+{
+    // This class contains utility methods that mimic the mscorlib internal System.Runtime.Versioning.BinaryCompatibility type.
+    internal sealed class BinaryCompatibility 
+    {
+        private static readonly Version Framework45 = new Version(4, 5);
+        private static readonly BinaryCompatibility Current = new BinaryCompatibility();
+
+        // Set the targetframework by:
+        // 1.   Looking at the targetFramework provided to: AppDomainSetup
+        // 2.   Failing that look at the Assemby executed with the AppDomain
+        // failing that then TargetsAtLeast_Desktop_V4_5 is set to true
+        public BinaryCompatibility()
+        {
+            string targetFrameworkName = null; // UNDONE UNDONE AppDomain.CurrentDomain.SetupInformation.TargetFrameworkName;
+            if(string.IsNullOrEmpty(targetFrameworkName))
+            {
+                targetFrameworkName = GetTargetFrameworkNameFromEntryAssembly();
+                if(string.IsNullOrEmpty(targetFrameworkName))
+                {
+                    TargetsAtLeast_Desktop_V4_5 = false;
+                    return;
+                }
+            }
+
+            var frameworkName = new FrameworkName(targetFrameworkName);
+            TargetsAtLeast_Desktop_V4_5 = (frameworkName.Version >= BinaryCompatibility.Framework45);
```

Comment:
Is this check even relevant on .NET Core?

Reply:
Good catch yes this can surely be hard coded to 'true' and delete this file entirely.